### ISSUE TITLE
Checking that <li> is the only direct child of <ul> and <ol> is too strict

### ISSUE THREAD

**Issue by apparebit** [jackdomleo7/Checka11y.css#69]
Checka11y.css currently flags any element but &lt;li&gt; that is directly nested within an unordered or ordered list as incorrectly placed. The documentation links to [Understanding WCAG 2.0](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html) but that page only provides high-level guidance.

The restriction on what elements may appear within lists is actually spelled out in the [HTML](https://html.spec.whatwg.org/multipage/grouping-content.html#the-ol-element) [standard](https://html.spec.whatwg.org/multipage/grouping-content.html#the-ul-element) already. However, it is a bit more permissive than Checka11y.css: In addition to &lt;li&gt;, it allows for script-supporting elements, that is &lt;script&gt; and &lt;template&gt;. It would be great if Checka11y.css allowed those two elements as well.

**Comment by jackdomleo7** [comment]
Hi, on those three links you provided, I'm not seeing any reference to `<script>` or `<template>` being a direct child of `<ol>` or `<ul>`. 🤔 Do you have any other links?

**Comment by apparebit** [comment]
If you follow the two links to the HTML Living Standard (which is the definitive document specifying HTML), the description includes the “Content model”. That’s what you are looking for. Under that entry, you’ll find &lt;li&gt; and “script-supporting elements”. The latter is a link and, if you click on it, you’ll find that those elements are the two other elements.

**Comment by Lewiscowles1986** [comment]
so just to check after reading the HTML living specification...

Only:

* li
* script
* template

can be children of

* ol
* ul

Is this the issue @apparebit?

**Comment by apparebit** [comment]
That's exactly it.

**Comment by Lewiscowles1986** [comment]
I'll see about making a PR. Should be relatively simple I think
